docs/source/gpu.rst

```diff
@@ -450,15 +466,90 @@ your GPU/TPU:
    G = jnp.array(np.random.normal(0, 1, (ny, nx)).astype(np.float32))
    x = jnp.ones(nx, dtype=np.float32)
 
+   # Create operator
    Gop = JaxOperator(MatrixMult(G, dtype='float32'))
-   y = Gop * x
+   
+   # Create data and invert
+   y = Gop @ x
    xest = Gop / y
 
    # Adjoint via AD
    xadj = Gop.rmatvecad(x, y)
 
+Again, the code is almost unchanged apart from the fact that we now use ``jax`` arrays.
+
+
+Mixed CPU-GPU powered inverse problems
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let us now consider a more intricate scenario where we have acess to 
+a GPU-powered operator, however the model and/or data vectors are too large 
+to fit onto the memory. 
+
+For the sake of clarity, we consider a problem where 
+the operator can be written as a :class:`pylops.BlockDiag` of
+PyLops operators. Note how, by simply sandwitching any of the GPU-powered 
+operator within two :class:`pylops.ToCupy` operators, we are
+able to tell PyLops to transfer to the GPU only the part of the model vector 
+required by a given operator and transfer back the output to the  CPU before 
+forming the combine output vector (i.e., the output vector of the 
+:class:`pylops.BlockDiag`)
+
+.. code-block:: python
+
+   nops, n = 5, 4
+   Ms = [np.diag((i + 1) * np.ones(n, dtype=dtype)) \
+            for i in range(nops)]
+   Ms = [M.T @ M for M in Ms]
+
+   # Create operator
+   Mops = []
+   for iop in range(nops):
+      Mop = MatrixMult(cp.asarray(Ms[iop], dtype=dtype))
+      Top = ToCupy(Mop.dims, dtype=dtype)
+      Top1 = ToCupy(Mop.dimsd, dtype=dtype)
+      Mop = Top1.H @ Mop @ Top
+      Mops.append(Mop)
+   Mops = BlockDiag(Mops, forceflat=True)
```

Comment:
I think a diagram showing all the operations and where they reside would be really useful. Something like this:
![to_cupy_diagram drawio](https://github.com/user-attachments/assets/3629a30b-383f-418d-b96c-51bc2bc00c7c)

(I can share the draw.io xml if you want).

Reply:
Thanks! I agree, it would be nice to have a few diagrams like this to explain the different behaviors 😊

Sure, can you please share it